_docs/tasks/traffic-management/egress-tcp.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Control Egress TCP Traffic
+overview: Describes how to configure Istio to route TCP traffic from services in the mesh to external services.
+
+order: 41
+
+layout: docs
+type: markdown
+---
+{% include home.html %}
+
+The [Control Egress Traffic]({{home}}/docs/tasks/traffic-management/egress.html) task demonstrated how external (out of the mesh) HTTP and HTTPS services can be accessed from applications inside the mesh. To remind: by default, Istio-enabled applications are unable to access URLs outside of the mesh. To enable such access, [egress rules]({{home}}/docs/reference/config/istio.routing.v1alpha1.html#EgressRule) must be defined, or, alternatively, [direct access to external services]({{home}}/docs/tasks/traffic-management/egress.html#calling-external-services-directly) must be configured.
+
+This task describes how to configure Istio to expose external TCP services to applications inside the Istio service mesh.
+
+## Before you begin
+
+* Setup Istio by following the instructions in the
+  [Installation guide]({{home}}/docs/setup/).
+
+* Start the [sleep](https://github.com/istio/istio/tree/master/samples/sleep) sample application which will be used as a test source for external calls.
+
+  ```bash
+  kubectl apply -f <(istioctl kube-inject -f samples/sleep/sleep.yaml)
+  ```
+
+  Note that any pod that you can execute `curl` from, would do.
+
+## Using Istio egress rules for external TCP traffic
+In this task we access `wikipedia.org` by HTTPS originated by the application. This task demonstrates the use case when the application cannot use HTTP with TLS origination by the sidecar proxy. Using HTTP with TLS origination by the sidecar proxy is described in the [Control Egress Traffic]({{home}}/docs/tasks/traffic-management/egress.html) task. In that task, `https://google.com` was accessed by issuing HTTP requests to  `http://www.google.com:443`.
+
+The HTTPS traffic  originated by the application will be treated by Istio as _opaque_ TCP. To enable such traffic, we define a TCP egress rule on port 443.
+
+In TCP egress rules as opposed to HTTP-based egress rules, the destinations are specified by IPs or by blocks of IPs in [CIDR notation](https://tools.ietf.org/html/rfc2317).
```

Comment:
Do we support using SNI hosts for egressing TCP connections if TLS is enabled?

Reply:
No, it is not yet supported by envoy https://github.com/envoyproxy/envoy/issues/1843